File: src/my_project_typescript/medium_problems_typescript/from51to100/wordBreak.ts
function wordBreak(s: string, wordDict: string[]): boolean {
    if(s.length < 1) return false; 
    const dp: boolean[] = new Array(s.length + 1).fill(false);
    dp[0] = true; 
    
    for(let i = 0; i < s.length; i++) {
        if(dp[s.length]) return dp[s.length]; 
        if(dp[i]) {
         for(let word of wordDict) {
            if(word === s.slice(i, i + word.length)) {
               dp[i + word.length] = true; 
            }
        }     
      }
    }

    return dp[s.length]
    
};

File: src/my_project/medium_problems/from101to150/word_break.py
from typing import List, Optional 

class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        d = [False] * len(s)    
        for i in range(len(s)):
            for w in wordDict:
                if w == s[i-len(w)+1:i+1] and (d[i-len(w)] or i-len(w) == -1):
                    d[i] = True
        print(d)
        return d[-1]
    
solution = Solution()

print(solution.wordBreak(s = "leetcode", wordDict = ["leet","code"]))